nginx.conf

```diff
@@ -0,0 +1,42 @@
+worker_processes  auto;
+error_log /dev/stdout info;
+
+events {
+    worker_connections  1024;
+}
+
+
+http {
+    include       /etc/nginx/mime.types;
+    default_type  application/octet-stream;
+
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log /dev/stdout;
+
+    sendfile        on;
+    
+    keepalive_timeout  65;
+
+    gzip  on;
+
+    server {
+        listen       80;
+        listen 443 default_server ssl;
+
+        server_name         github.com;
+        ssl_certificate /etc/ssl/certs/github.com.crt;
+        ssl_certificate_key /etc/ssl/private/github.com.key;
+
+        location /istio/istio/releases/download/ {
+            alias   /opt/istio-releases/;
```

Comment:
So this location catches everything at that URL and then redirects it to the folder, and would return the tars that exist there. What happens if a tar doesn't exist, nginx 404's and it fails like it does now ?

Reply:
Exactly! 

Here is an example of the error you would see inside of the istio-installer pod logs: 
```
2021/09/21 22:38:32 [error] 26#26: *1 open() "/opt/istio-releases/1.9.8/istio-1.9.8-linux-amd64.tar.gz" failed (2: No such file or directory), client: 127.0.0.1, server: github.com, request: "GET /istio/istio/releases/download/1.9.8/istio-1.9.8-linux-amd64.tar.gz HTTP/1.1", host: "github.com"
127.0.0.1 - - [21/Sep/2021:22:38:32 +0000] "GET /istio/istio/releases/download/1.9.8/istio-1.9.8-linux-amd64.tar.gz HTTP/1.1" 404 125 "-" "Go-http-client/1.1"
2021/09/21 22:38:32 [info] 26#26: *1 client 127.0.0.1 closed keepalive connection
2021-09-21T22:38:32.205936Z	info	Error: failed to generate Istio configs from file [/app/istio-base.yaml] for the current version: 1.9.8, error: failed to fetch URL https://github.com/istio/istio/releases/download/1.9.8/istio-1.9.8-linux-amd64.tar.gz : 404 Not Found

Error: failed to generate Istio configs from file [/app/istio-base.yaml] for the current version: 1.9.8, error: failed to fetch URL https://github.com/istio/istio/releases/download/1.9.8/istio-1.9.8-linux-amd64.tar.gz : 404 Not Found
error found during istioctl upgrade
```
And the helm error returns: 
```
Error: UPGRADE FAILED: post-upgrade hooks failed: job failed: BackoffLimitExceeded
```

If for some reason nginx fails to spin up, you would see a 443 error, and it would act the same way.

Here is an example of nginx not being on, but traffic was directed to it:
```
Upgrade version check passed: 1.10.4 -> 1.10.4.

2021-09-21T20:45:10.470612Z	info	Error: failed to generate Istio configs from file [/app/istio-base.yaml] for the current version: 1.10.4, error: Get "https://github.com/istio/istio/releases/download/1.10.4/istio-1.10.4-linux-amd64.tar.gz": dial tcp 127.0.0.1:443: connect: connection refused

Error: failed to generate Istio configs from file [/app/istio-base.yaml] for the current version: 1.10.4, error: Get "https://github.com/istio/istio/releases/download/1.10.4/istio-1.10.4-linux-amd64.tar.gz": dial tcp 127.0.0.1:443: connect: connection refused
error found during istioctl upgrade
```

And the helm error returns: 
```
Error: UPGRADE FAILED: post-upgrade hooks failed: job failed: BackoffLimitExceeded
```